token-lending/program/tests/redeem_fees.rs

```diff
@@ -0,0 +1,281 @@
+                solend_program::id(),
+                sol_test_reserve.pubkey,
+                sol_oracle.pyth_price_pubkey,
+                sol_oracle.switchboard_feed_pubkey,
+            ),
+        ],
+        Some(&payer.pubkey()),
+    );
+
+    transaction.sign(&[&payer], recent_blockhash);
+    assert!(banks_client.process_transaction(transaction).await.is_ok());
+
+    let sol_reserve_before = sol_test_reserve.get_state(&mut banks_client).await;
+    let usdc_reserve_before = usdc_test_reserve.get_state(&mut banks_client).await;
+    let sol_balance_before =
+        get_token_balance(&mut banks_client, sol_reserve_before.config.fee_receiver).await;
+    let usdc_balance_before =
+        get_token_balance(&mut banks_client, usdc_reserve_before.config.fee_receiver).await;
+
+    let mut transaction2 = Transaction::new_with_payer(
+        &[
+            redeem_fees(
+                solend_program::id(),
+                usdc_test_reserve.pubkey,
+                usdc_test_reserve.config.fee_receiver,
+                usdc_test_reserve.liquidity_supply_pubkey,
+                lending_market.pubkey,
+            ),
+            redeem_fees(
+                solend_program::id(),
+                sol_test_reserve.pubkey,
+                sol_test_reserve.config.fee_receiver,
+                sol_test_reserve.liquidity_supply_pubkey,
+                lending_market.pubkey,
+            ),
+        ],
+        Some(&payer.pubkey()),
+    );
+
+    transaction2.sign(&[&payer], recent_blockhash);
+    assert!(banks_client.process_transaction(transaction2).await.is_ok());
+
+    let sol_reserve = sol_test_reserve.get_state(&mut banks_client).await;
+    let usdc_reserve = usdc_test_reserve.get_state(&mut banks_client).await;
+    let sol_balance_after =
+        get_token_balance(&mut banks_client, sol_reserve.config.fee_receiver).await;
+    let usdc_balance_after =
+        get_token_balance(&mut banks_client, usdc_reserve.config.fee_receiver).await;
+
+    let slot_rate = Rate::from_percent(BORROW_RATE)
+        .try_div(SLOTS_PER_YEAR)
+        .unwrap();
+    let compound_rate = Rate::one()
+        .try_add(slot_rate)
+        .unwrap()
+        .try_pow(SLOTS_ELAPSED)
+        .unwrap();
+    let compound_borrow = Decimal::from(BORROW_AMOUNT).try_mul(compound_rate).unwrap();
+
+    let net_new_debt = compound_borrow
+        .try_sub(Decimal::from(BORROW_AMOUNT))
+        .unwrap();
+    let protocol_take_rate = Rate::from_percent(sol_test_reserve.config.protocol_take_rate);
+    let delta_accumulated_protocol_fees = net_new_debt.try_mul(protocol_take_rate).unwrap();
+    let delta_borrowed_amount_wads = net_new_debt
+        .try_sub(delta_accumulated_protocol_fees)
+        .unwrap();
+    let new_borrow_amount_wads = Decimal::from(BORROW_AMOUNT)
+        .try_add(delta_borrowed_amount_wads)
+        .unwrap()
+        .try_add(Decimal::from(usdc_balance_after - usdc_balance_before))
+        .unwrap();
+
+    assert_eq!(
```

Comment:
these first 6 assertions are interesting they assert that total supply and exchange rate dont change when claiming fees (which makes sense) 

but utilization does (because we take from available liquidity and add to debt. this balances the accounting that occurs over time where borrowers were accumulating more debt than lenders were owed until fees are claimed